### PR TITLE
[CSM] Block access to the `io` library

### PR DESCRIPTION
**Pull request by red-001**

### CODE REVIEW

**Comment by ShadowNinja** [comment]
No need to add the `io` table at all if you remove everything in it.

In the future we may want to add some of these back and just allow mods to access special files through Minetest APIs, but I'd rather not have direct filesystem access ever available (the current checkPath method is a compatibility hack and it's quite easy to accidentally make the wrong files accessible.  I think it would be best to have something like core.get_mod_file("foo.txt") which creates a file in an implementation-defined location (like `~/.minetest/client/mods/$mod/foo.txt`) and returns the file.  This would even allow you to transparently store all mod data in a database instead of on the filesystem).

This looks good, although I'd completely remove the `io` table instead of just emptying it.

**Comment by nerzhul** [comment]
@red-001 can you fix it like @ShadowNinja suggests ?

File: builtin/common/misc_helpers.lua [hunk 1]
```diff
@@ -197,16 +197,17 @@ assert(table.indexof({"foo", "bar"}, "foo") == 1)
...
 
 --------------------------------------------------------------------------------
-function file_exists(filename)
-	local f = io.open(filename, "r")
-	if f == nil then
-		return false
-	else
-		f:close()
-		return true
+if INIT ~= "client" then
```
Comment (nerzhul) [inline builtin/common/misc_helpers.lua]: why not move this file in another common lua file and call it from misc_helpers callers (except client) ?

File: src/script/cpp_api/s_security.cpp [hunk 2]
```diff
@@ -339,6 +320,7 @@ void ScriptApiSecurity::initializeSecurityClient()
 	lua_pop(L, 1);  // Pop old jit
 #endif
 
+
```
Comment (nerzhul) [inline src/script/cpp_api/s_security.cpp]: no need for this line

**Comment by red-001** [comment]
Done

File: builtin/common/misc_helpers.lua (hunk 1 above)
Comment (red-001) [inline builtin/common/misc_helpers.lua]: checking INIT is already done a lot in `misc_helpers`, not sure if fixing that shouldn't be done in a separate PR.

File: src/script/cpp_api/s_security.cpp (hunk 2 above)
Comment (red-001) [inline src/script/cpp_api/s_security.cpp]: removed

**Comment by nerzhul** [comment]
merged, thanks.

@est31 you are back ? if i remember you said you are not interested in minetest anymore last week ?

**Comment by est31** [comment]
@nerzhul no I'm still gone. I just saw this PR after I was ranting about the situation a few days ago :). Its an exception :)

**Comment by nerzhul** [comment]
heh, if CSM interest you, don't hesitate to review, but as i see you are very busy with RUST :p
